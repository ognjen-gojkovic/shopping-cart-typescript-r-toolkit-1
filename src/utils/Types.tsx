export type Product = {
  brand: string;
  category: string;
  description: string;
  discountPrecentage: number;
  id: number;
  images: string[];
  price: number;
  rating: number;
  stock: number;
  thumbnail: string;
  title: string;
};

export type CartItem = {
  brand: string;
  category: string;
  description: string;
  discountPrecentage: number;
  id: number;
  images: string[];
  price: number;
  rating: number;
  stock: number;
  thumbnail: string;
  title: string;
  qty: number;
};
