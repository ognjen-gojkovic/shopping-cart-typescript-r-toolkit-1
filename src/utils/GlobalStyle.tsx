import { css, createGlobalStyle } from "styled-components";

const Reset = css`
  // variables
  :root {
  }

  *,
  *::before,
  *::after {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: sans-serif;
  }
`;

const BaseStyle = createGlobalStyle`
    ${Reset}
`;

export const GlobalStyle = () => <BaseStyle />;
