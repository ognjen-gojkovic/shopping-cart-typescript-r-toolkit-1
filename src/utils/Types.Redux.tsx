import { INITIAL_STATE } from "../redux/reducers/Reducer.Products";
import { Product } from "./Types";

export type ResObjType = {
  status: boolean;
  products: Product[];
  error: object | null;
};

export type rootState = {
  ReducerProducts: INITIAL_STATE;
};
