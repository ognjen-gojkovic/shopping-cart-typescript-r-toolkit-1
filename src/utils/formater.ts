//export const fromatText = <>(T extends string): T => {};

export function formatText<T extends string>(text: T): string {
  let temp = text.slice(0, 45);
  if (temp[temp.length - 1] === " " || temp[temp.length - 1] === ".") {
    let arr = temp.split("");
    arr.pop();
    temp = arr.join("");
  }
  let newText: string = temp.concat("...");

  return newText;
}
