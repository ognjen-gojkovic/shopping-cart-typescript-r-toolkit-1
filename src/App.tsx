import React from "react";
import { useAppDispatch, useAppSelector } from "./redux/hooks/reduxHooks";
import { fetchAllProducts } from "./redux/actions/Actions.Products";
import styled from "styled-components";
import { GlobalStyle } from "./utils/GlobalStyle";
import Navbar from "./components/navbar/Navbar";
import Home from "./components/home/Home";
import Cart from "./components/cart/Cart";

function App(): React.ReactElement {
  const dispatch = useAppDispatch();
  const [showCart, setShowCart] = React.useState<boolean>(false);
  const cart = useAppSelector((state) => state.reducerCart);

  React.useEffect(() => {
    dispatch(fetchAllProducts());

    if (cart.cart.length > 0)
      localStorage.setItem("cart", JSON.stringify(cart));
  }, [dispatch, cart]);

  const handleCartToggle = (from?: string): void => {
    if (from === "navbar") setShowCart(true);
    else setShowCart(false);
  };

  return (
    <AppStyled className="App">
      <GlobalStyle />
      <Navbar handleCartToggle={handleCartToggle} />
      <Cart toggle={showCart} handleToggleCart={handleCartToggle} />
      <Home />
    </AppStyled>
  );
}

const AppStyled = styled.div`
  width: 100%;
  height: 100%;
  background: linear-gradient(135deg, #00c3ff, #3a5eff);
`;

export default App;
