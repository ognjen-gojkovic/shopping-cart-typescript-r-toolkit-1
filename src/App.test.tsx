import { screen, cleanup } from "@testing-library/react";
import App from "./App";
import UserEvent from "@testing-library/user-event";
import { renderWithProviders } from "./utils/renderWithRedux";

describe("App component", () => {
  afterEach(cleanup);
  test("should correctly render app tree", () => {
    const { container } = renderWithProviders(<App />);

    expect(container).toMatchSnapshot();
  });

  test("should correctly change cart show state", async () => {
    renderWithProviders(<App />);

    const cart = screen.getByTestId("cart-component");
    expect(cart).toHaveStyle("opacity: 0");

    const btn = screen.getByTestId("cart");
    await UserEvent.click(btn);

    expect(cart).toHaveStyle("opacity: 1");
  });
});
