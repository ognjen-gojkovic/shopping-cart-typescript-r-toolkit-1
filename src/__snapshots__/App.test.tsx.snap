// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should correctly render app tree 1`] = `
<div>
  <div
    class="sc-eDvSVe gzbpHj App"
  >
    <nav
      class="sc-bcXHqe eReuuJ"
    >
      <h2>
        Shopping cart app
      </h2>
      <div
        class="container"
      >
        <span>
          Total: $
          0
        </span>
        <div
          class="cart"
          data-testid="cart"
        >
          <i />
           Cart
        </div>
      </div>
    </nav>
    <div
      class="sc-hLBbgP eePoFQ"
      data-testid="cart-component"
    >
      <div
        class="modal"
      />
      <div
        class="cart"
      >
        <h2>
          Your Items
        </h2>
        <h2>
          Cart is empty.
        </h2>
      </div>
    </div>
    <div
      class="sc-dkrFOg CONCD"
    >
      <h2>
        Products
      </h2>
      <div
        class="products"
      />
    </div>
  </div>
</div>
`;

exports[`redux test should first 1`] = `
<div>
  <div
    class="sc-eDvSVe gzbpHj App"
  >
    <nav
      class="sc-bcXHqe eReuuJ"
    >
      <h2>
        Shopping cart app
      </h2>
      <div
        class="container"
      >
        <span>
          Total: $
          0
        </span>
        <div
          class="cart"
          data-testid="cart"
        >
          <i />
           Cart
        </div>
      </div>
    </nav>
    <div
      class="sc-hLBbgP eePoFQ"
    >
      <div
        class="modal"
      />
      <div
        class="cart"
      >
        <h2>
          Your Items
        </h2>
        <h2>
          Cart is empty.
        </h2>
      </div>
    </div>
    <div
      class="sc-dkrFOg CONCD"
    >
      <h2>
        Products
      </h2>
      <div
        class="products"
      />
    </div>
  </div>
</div>
`;
